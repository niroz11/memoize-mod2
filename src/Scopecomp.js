import React, { Component } from "react";
import Card from "./Card";
import Instruction from "./Instruction";

class Scopecomp extends Component {
  render() {
    const { data, question, score } = this.props;

    
    return (
      <div className="proto">
        {question === -1 && (
          <Instruction
            increment={this.props.increment}
            toggleCard={this.props.toggleCard}
            showCard={this.props.showCard}
          />
        )}
        {question > -1 && question < data.length && (
          <Card
            question={data[question]}
            increment={this.props.increment}
            score={score}
            updateScore={this.props.updateScore}
            saveLocalStorage={this.props.localStorage}
          />
        )}

        {question > data.length - 1 && <p>{score}</p> && this.props.mainMenu()}
      </div>
    );
  }
}

export default Scopecomp;
