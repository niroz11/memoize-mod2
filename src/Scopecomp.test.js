import React from "react";
import Scopecomp from "./Scopecomp";
import { shallow } from "enzyme";



describe("Scopecomp", () => {
  

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Scopecomp
              data={clickValue === "quiz" ? data : incorrect}
              question={this.state.question}
              increment={this.incrementQuestion}
              score={this.state.score}
              updateScore={this.updateScore}
              mainMenu={this.getMainMenu}
              localStorage={this.saveToLocalStorage}
              toggleCard={this.toggleCard}
              showCard={this.state.showCard}
            />
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  

  
});

