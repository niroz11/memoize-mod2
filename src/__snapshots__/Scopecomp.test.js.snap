// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scopecomp should match the snapshot with all data passed in 1`] = `
<div
  className="proto"
>
  <Instruction
    increment={[MockFunction]}
    showCard={true}
    toggleCard={[MockFunction]}
  />
</div>
`;
